File: src/Array/cpp/qiuyi/RemoveDuplicatesfromSortedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
    if (head == NULL || head->next == NULL) return head;
    
    struct ListNode *cur, *tmp, *p;
    cur = head;
    
    while (cur != NULL) {
        tmp = cur->next;
        while (tmp != NULL && cur->val == tmp->val) {
            p = tmp;
            tmp = tmp->next;
            free(p);
        }
        cur->next = tmp;
        cur = cur->next;
    }
    
    return head;
    }
};

File: src/haoxiang/LinkedList/RemoveDuplicatesfromSortedList.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        pos = head
        while pos:
            while pos.next and pos.val == pos.next.val:
                pos.next = pos.next.next
            pos = pos.next
        return head
